90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -2099,7 +2126,16 @@ emergency_shell() {
 
   echo -n "Launching emergency shell: "
   echo -e "${message}\n"
-  /bin/bash -l
+
+  # -i (interactive) mode will source /.bashrc
+  /bin/bash -i
+
+  while read -r _ mp fs _ ; do
+    if [ "${fs}" == "zfs" ]; then
+      umount "${mp}"
+      zdebug "unmounting: ${mp}"
+    fi
+  done < /etc/mtab
```

Comment:
On my system, `/etc/mtab` is just a symlink to `/proc/self/mounts`. There is also a `/proc/mounts` that links to `/proc/self/mounts`.

I wonder whether there are cases on modern systems where `/etc/mtab` is a static file and whether we should prefer `/proc/mounts` or `/proc/self/mounts`, maybe falling through a chain of preferences if one can't be found.

Reply:
This is probably a more universal way to ensure access to an `mtab` file. I've also removed usage of `_` as a dummy variable, `$_` is apparently a bash special variable.